libclamav/dlp.c

```diff
@@ -128,42 +128,43 @@ static int ssn_max_group[MAX_AREA+1] = { 0,
 struct iin_map_struct {
     uint32_t iin_start;
     uint32_t iin_end;
-    uint8_t card_len;
+    uint8_t card_min;
+    uint8_t card_max;
+    uint8_t is_cc;
     uint8_t luhn;
     const char* iin_name;
 };
 
-/* Maestro card range, 550000-699999, encompasses ranges
-   of several other cards including Discover and UnionPay.
-*/
-
 static struct iin_map_struct iin_map[] = {
-    {100000, 199999, 15, 1, "UATP"},
-    {300000, 305999, 14, 1, "Diner's Club - Carte Blanche"},
-    {309500, 309599, 14, 1, "Diner's Club International"},
-    {340000, 349999, 15, 1, "American Express"},
-    {352800, 358999, 16, 1, "JCB"},
-    {360000, 369999, 14, 1, "Diner's Club International"},
-    {370000, 379999, 15, 1, "American Express"},
-    {380000, 399999, 16, 1, "Diner's Club International"},
-    {400000, 499999, 16, 1, "Visa"},
-    {500000, 509999, 16, 1, "Maestro"},
-    {510000, 559999, 16, 1, "Master Card"},
-    {560000, 699999, 16, 1, "Maestro/Discover/UnionPay/etc."},
+    {100000, 199999, 13, 15, 1, 0, "UATP"},
+    {222100, 272099, 16, 16, 1, 1, "Mastercard 2016"},
+    {300000, 305999, 14, 16, 1, 1, "Diner's Club - Carte Blanche"},
+    {309500, 309599, 14, 16, 1, 1, "Diner's Club International"},
+    {340000, 349999, 15, 15, 1, 1, "American Express"},
+    {352800, 358999, 16, 16, 1, 1, "JCB"},
+    {360000, 369999, 14, 16, 1, 1, "Diner's Club International"},
+    {370000, 379999, 15, 15, 1, 1, "American Express"},
+    {380000, 399999, 16, 16, 1, 1, "Diner's Club International"},
+    {400000, 499999, 16, 16, 1, 1, "Visa"},
+    {500000, 509999, 16, 16, 1, 0, "Maestro"},
```

Comment:
The `0` should be the is_cc field? Isn't the 6th field in iin_map_struct the `luhn` field?